lib/src/conflicts.rs

```diff
@@ -302,6 +312,26 @@ fn materialize_conflict_hunks(
     Ok(())
 }
 
+fn materialize_git_diff3_conflict(
```

Comment:
Right now, it's a bit unusual that the "git"/"diff3" style labels the "base" section but not either of the sides' sections, so I'm wondering if instead of this format:

```
<<<<<<< Conflict 1 of 1
fn example(word: String) {
    println!("word is {word}");
||||||| Contents of base
fn example(w: String) {
    println!("word is {w}");
=======
fn example(w: &str) {
    println!("word is {w}");
>>>>>>> Conflict 1 of 1 ends
}
```

We should have this format (inspired by @ilyagr's comment):

```
<<<<<<< Side #1 (Conflict 1 of 1)
fn example(word: String) {
    println!("word is {word}");
||||||| Base
fn example(w: String) {
    println!("word is {w}");
=======
fn example(w: &str) {
    println!("word is {w}");
>>>>>>> Side #2 (Conflict 1 of 1 ends)
}
```

Thoughts?

Reply:
Weak vote for the second format.

Btw, if we plan to support more than 2 sides, then the second side will have to go on the `=======` line instead of on the `>>>>>>>` line (yes, different from Git). I guess there's not much reason to support more than 2 sides using this format, and the fact that the labels with more than 2 sides will either be inconsistent with Git or with how they look for 2-sides conflicts seems like enough of a reason to say that we simply won't support more than 2 sides ever (falling back to "snapshot" instead).

Reply:
> Btw, if we plan to support more than 2 sides, then the second side will have to go on the `=======` line instead of on the `>>>>>>>` line (yes, different from Git). I guess there's not much reason to support more than 2 sides using this format, and the fact that the labels with more than 2 sides will either be inconsistent with Git or with how they look for 2-sides conflicts seems like enough of a reason to say that we simply won't support more than 2 sides ever (falling back to "snapshot" instead).

Yeah, I agree with all of that. I updated it to the second format since it matches the Git format a bit better, and it seems like the point of having this option is to match Git as closely as possible. I also updated the fallback to "snapshot" since it's more similar of a format.

Reply:
If we really want to make this format compatible with Git tools, we might want to eventually have a "yours:filename" and "ours:filename" side, like https://git-scm.com/docs/git-merge#_how_conflicts_are_presented . I have no idea how many of Git tools rely on those markers being in that format, we could first check whether anybody requests it.

For now it'd be difficult to consistently pick an ours/theirs side, we'd have to pick arbitrarily. So, there's less reason to follow Git here.

As long as we're not following Git, I also think the second format is a bit nicer.

Reply:
I think the format actually depends on the version of Git, the user's config options, and/or which particular command produced the conflict, because this is what I get when running `git merge branch-name` on my machine:

```
<<<<<<< HEAD
fn example(word: String) {
    println!("word is {word}");
|||||||
fn example(w: String) {
    println!("word is {w}");
=======
fn example(w: &str) {
    println!("word is {w}");
>>>>>>> branch-name
}
```

For `git rebase branch-name`, the conflict end marker is slightly different, with `<short commit ID> (<description>)` for the text. Anecdotally, I've seen Git produce a few different formats of text next to these markers, so I don't think we should be too concerned about matching exactly. If someone runs into a case where a tool doesn't parse our format, we can always update it later.